Transform `COS` plugin into `Observability` plugin

sudeephb opened the pull request:
- Rename `COS` plugin to `Observability` plugin
- Add terraform configuration for grafana-agent and its relations

hemanthnakkina commented on sunbeam-python/sunbeam/plugins/observability/etc/deploy-grafana-agent/main.tf:
```diff
@@ -0,0 +1,97 @@
+}
+
+# juju integrate openstack-hypervisor:cos-agent grafana-agent:cos-agent
+resource "juju_integration" "openstack-hypervisor-to-grafana-agent" {
```
can you make this generic instead of using wording "openstack-hypervisor". The reason being grafana-agent terraform module can be generic and can be reusable.

hemanthnakkina commented on sunbeam-python/sunbeam/plugins/observability/etc/deploy-grafana-agent/variables.tf:
```diff
@@ -0,0 +1,41 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+variable "openstack-hypervisor-name" {
```
see the comment in main.tf

hemanthnakkina commented on sunbeam-python/sunbeam/plugins/observability/plugin.py:
```diff
@@ -49,26 +49,33 @@
 LOG = logging.getLogger(__name__)
 console = Console()
 
 COS_MODEL = "cos"
```
Can we change the model name to observability

hemanthnakkina commented on sunbeam-python/sunbeam/plugins/observability/plugin.py:
```diff
@@ -49,26 +49,33 @@
 
-class EnableCosStep(BaseStep, JujuStepHelper):
-    """Deploy COS using Terraform"""
+class DeployCosStep(BaseStep, JujuStepHelper):
```
Use Observability or observability stack wherever applicable instead of cos

hemanthnakkina commented on sunbeam-python/sunbeam/plugins/observability/etc/deploy-grafana-agent/variables.tf:
```diff
@@ -0,0 +1,40 @@
+}
+
+variable "controller-model" {
+  description = "Name of the sunbeam controller model"
```
Change to "Name of the model principal application deployed in"

hemanthnakkina commented on sunbeam-python/sunbeam/plugins/observability/etc/deploy-grafana-agent/variables.tf:
```diff
@@ -0,0 +1,40 @@
+
+variable "grafana-agent-channel" {
+  description = "Channel to use when deploying grafana agent machine charm"
+  default = "latest/candidate"
```
Can you add a comment above that channel latest/stable not available for grafana-agent and so defaulted to latest/candidate

hemanthnakkina commented on sunbeam-python/sunbeam/plugins/observability/plugin.py:
```diff
@@ -0,0 +1,455 @@
+        jhelper: JujuHelper,
+    ):
+        super().__init__(
+            "Deploying Observability Stack", "Deploying Observability Stack"
```
"Deploy Observability Stack", "Deploying Observability Stack"

hemanthnakkina commented on sunbeam-python/sunbeam/plugins/observability/plugin.py:
```diff
@@ -0,0 +1,455 @@
+        tfhelper_cos: TerraformHelper,
+        jhelper: JujuHelper,
+    ):
+        super().__init__("Deploying Grafana Agent", "Deploying Grafana Agent")
```
"Deploy Grafana Agent", "Deploying Grafana Agent"

hemanthnakkina commented on sunbeam-python/sunbeam/plugins/observability/plugin.py:
```diff
@@ -0,0 +1,455 @@
+
+        cos_backend = self.tfhelper_cos.backend
+        cos_backend_config = self.tfhelper_cos.backend_config()
+        config = self.read_config()
```
Need to catch ConfigItemNotFoundException. Replicate in other places where read_config() is used

hemanthnakkina commented on sunbeam-python/sunbeam/plugins/observability/plugin.py:
```diff
@@ -0,0 +1,455 @@
+        self.tfhelper_cos = tfhelper_cos
+        self.jhelper = jhelper
+        self.model = CONTROLLER_MODEL
+        self.read_config = lambda: plugin.get_plugin_info().get("config", {})
```
This is overriding the TeraformVars for COS saved in cluster database. Use different keys for COS and grafana-agent terraform plans.

hemanthnakkina reviewed (changes requested):
Please see comments inline - the major one being using same key to save terraform vars fro cos and grafana-agent in DB which results in overwrites.

Also I see no Plan to enable monitoring on openstack-hypervisor. Can you include that step.

sudeephb commented:
> Please see comments inline - the major one being using same key to save terraform vars fro cos and grafana-agent in DB which results in overwrites.

Thank you, I have addressed the comments.

> Also I see no Plan to enable monitoring on openstack-hypervisor. Can you include that step.

This is not required since the plan for grafana-agent enables the relation, and when the relation is enabled, the charm handles turning the exporter services on.

hemanthnakkina reviewed (approved):
LGTM

hemanthnakkina commented:
@sudeephb Can you sign your commits?
